### DISCUSSION TITLE
Not referencing local repository for dependency

### DISCUSSION THREAD

**Discussion by dudeNumber4** [jbangdev/jbang#1663] [Q&A]
The [doc](https://www.jbang.dev/documentation/guide/latest/dependencies.html#repositories) reads, "By default, jbang uses ~/.m2 as local repository," but I'm not finding that to be the case.

In my java file I have:
//DEPS sap.jdbc4.sqlanywhere:sajdbc4:17.0.11.7173

In my local repository I have:
// C:\Users\me\.m2\repository\sap\jdbc4\sqlanywhere\sajdbc4\17.0.11.7173

When I run jbang against my file I get:
[jbang] Resolving dependencies...
[jbang]    sap.jdbc4.sqlanywhere:sajdbc4:17.0.11.7173
[jbang] [ERROR] Could not resolve dependencies: Could not find artifact sap.jdbc4.sqlanywhere:sajdbc4:jar:17.0.11.7173 in mavencentral (https://repo1.maven.org/maven2/)

What do I need to do to make it reference the local repo?

**Comment by maxandersen** [comment]
Are you using latest jbang ?

Your usecase is you have a jar locally built that is not in any of the main repositories setup?

It *should* work but could be it got lost as I just realize we don't seem to have a test for that specific use case. 

A workaround is to do `//REPOS local=pathtolocal`

But yeah we should consider adding local and central where local is the configured  download location. 

Today we just add central.

**Reply by dudeNumber4** [reply]
- Version reports 0.108.0
- Yes, jar is not hosted in any public repository that I can find, but I didn't build it.
- `//REPOS local=` with any of these paths (~/.m2, C:\Users\me\.m2, C:/Users/me/.m2, ~/.m2/repository) gives this partial error stack.  I'm attempting from an elevated prompt; should not be a permissions issue.  The path in my local repo in my original post lacked a '\', the path I used now is correct.

Could not transfer artifact sap.jdbc4.sqlanywhere:sajdbc4:pom:17.0.11.7173 from/to local (~/.m2): Cannot access ~/.m2 with type default using the available connector factories
Cannot access ~/.m2 with type default using the available connector factories: BasicRepositoryConnectorFactory
Cannot access ~/.m2 using the registered transporter factories: HttpTransporterFactory, FileTransporterFactory

**Reply by maxandersen** [reply]
Might need to be a url:


file:///Users/...

**Reply by dudeNumber4** [reply]
Error: _Could not find artifact sap.jdbc4.sqlanywhere:sajdbc4:jar:17.0.11.7173 in local (file:///C:\Users\me\.m2)_.  Same with forward slashes.

Let me try something simpler.  Why can't I simply pass cp/classpath?  As I tried [here](https://stackoverflow.com/questions/76576834/this-simple-sample-runs-in-jshell-but-not-in-jbang-what-could-cause-this-differ) (before I got the memo that stackoverflow isn't used anymore)?

**Reply by dudeNumber4** [reply]
The above case turned out just to be spaces in the class path.  Still don't know what the issue with local repository is.

**Reply by maxandersen** [reply]
cool - glad you got it solved. spaces in paths and terminals is always "fun".

about local repo I'm thinking it could just be that if you only placed the jar inside the repo you don't have the accompanying necessary metadata files that a local maven repo have?

**Reply by dudeNumber4** [reply]
I pulled the dependency using maven get, so it should be complete.  The accompanying pom, sha files are present.
